ICSharpCode.Decompiler/IL/Transforms/InlineExpressionTreeParameterDeclarations.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Diagnostics;
+using System.Linq;
+using ICSharpCode.Decompiler.TypeSystem;
+
+namespace ICSharpCode.Decompiler.IL.Transforms
+{
+	class InlineExpressionTreeParameterDeclarationsTransform : IStatementTransform
+	{
+		public void Run(Block block, int pos, StatementTransformContext context)
+		{
+			if (block.Instructions[pos] is StLoc parameterVariableInst) {
+				bool alwaysInlined = true;
+				if (MatchParameterVariableAssignment(parameterVariableInst, out ILVariable v, out ILInstruction init)) {
+					foreach (var LdParameterVariableInst in context.Function.Descendants.OfType<LdLoc>()) {
+						if (LdParameterVariableInst.MatchLdLoc(v)) {
```

Comment:
`v.LoadInstructions` will give you all the LdLocs for a variable, so you don't need to search the whole function.

Reply:
Concurrent modification of the list - does not really work...

Reply:
Feel free to use `ToArray()` to create a temporary copy.